ggml-cuda.cu

```diff
@@ -1823,6 +1963,29 @@ static __global__ void dequantize_block_iq3_xxs(const void * __restrict__ vx, ds
 
 }
 
+template<typename dst_t>
+static __global__ void dequantize_block_iq1_s(const void * __restrict__ vx, dst_t * __restrict__ yy) {
+
+    const int i   = blockIdx.x;
+    const block_iq1_s * x = (const block_iq1_s  *) vx;
+
+    const int tid = threadIdx.x;
+#if QK_K == 256
+    const int il = tid/8; // 0...3
+    const int ib = tid%8; // 0...7
+    dst_t * y = yy + i*QK_K + 32*ib + 8*il;
```

Comment:
If this kernel is in a critical path for performance, it may help performance a bit if we ensure that `y` is aligned on 16-byte boundary (if that's that's the case). Right now all results are stored in `y` one by one and for this relatively low-compute kernel memory i/o will likely be the bottleneck. Aligned pointer would allow compiler to perform the store as a single 128-bit operation.

https://cuda.godbolt.org/z/a1Eczhv5a

<img width="843" alt="image" src="https://github.com/ggerganov/llama.cpp/assets/526795/0b4a8a45-29f4-497a-84d8-9f733b393f0d">